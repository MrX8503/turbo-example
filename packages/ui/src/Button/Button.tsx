import type { ReactNode } from 'react';

export interface ButtonProps {
    className?: string;
    children?: ReactNode;
}

export default function Button ({ children, className }: ButtonProps) {
    return (
        <button
            type="button"
            className={`ui-text-red-800 ui-bg-slate-700 ${className || ''}`}
        >
            {children}
        </button>
    );
}
