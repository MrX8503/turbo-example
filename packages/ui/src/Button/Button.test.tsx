import renderer from 'react-test-renderer';

import Button from './Button';

describe('Button', () => {
    it('snapshot is unchanged', () => {
        const toRender = (
            <Button>
                Hello WOrld
            </Button>
        );

        const tree = renderer.create(toRender).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
