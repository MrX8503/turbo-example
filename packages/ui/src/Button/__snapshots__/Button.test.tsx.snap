// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshot is unchanged 1`] = `
<button
  className="ui-text-red-800 ui-bg-slate-700 "
  type="button"
>
  Hello WOrld
</button>
`;
